Microsoft.Packaging.Tools/docs/trimming.md

```diff
@@ -0,0 +1,73 @@
+
+Examples:
+```
+dotnet build /p:TrimUnusedDependencies=true
+dotnet publish /p:TrimUnusedDependencies=true
+msbuild /p:TrimUnusedDependencies=true
+msbuild /t:Publish /p:TrimUnusedDependencies=true
+```
+
+### From the IDE or committing the change to your project
+
+In your project (*.csproj* file) make the following change.
+
+```xml
+<PropertyGroup>
+  <TrimUnusedDependencies>true</TrimUnusedDependencies>
+</PropertyGroup>
+```
+
+### Additional options
+`@(TrimFilesRootFiles)` -  Additional *root* files to consider part of the application.  See [roots](#roots).  
+`@(TrimFilesRootPackages)` -  Additional *root* packages to consider part of the application.  See [roots](#roots).  
+`@(TrimmableFiles)` - Files which should be trimmed from the application.  See [trimmable](#trimmable).  
+`@(TrimmablePackages)` - Packages which should be trimmed from the application.  See [trimmable](#trimmable).  
+`$(TrimFilesPreferNativeImages)` - Prefer a file with the `.ni.dll` extension over a file with the `.dll` extension.  `.ni.dll` files are native images and significantly larger than a managed assembly but will load faster since they don't need to be JIT compiled.
+
+## How it works
+The trimming task examines all of the binaries and packages that make up your project and constructs a graph of the two that is related.  We start by identifying roots that are included in the application then we traverse the relationships between those to determine if other files or packages should be included in the app.
+
+### Roots
+By default the application is a *root*, as well as all `PackageReference`s from the project file.
+
+The direct packages references may be excluded from the set of *roots* by specifying the property `RootPackageReference=false`.
+
+Additional file *roots* may be specified using the `TrimFilesRootFiles` item.
+Additional package *roots* may be specified using the `TrimFilesRootPackages` item.
+
+### Trimmable
+Files or packages may be treated as *trimmable*.  Essentially this means that when the file or package is encountered during graph traversal, traversal will halt down that path.
+
+If a file is *trimmable* this means that the file will not be included in the application.  A reference to the file alone will not cause the file to be included, nor will a direct reference to a package containing that file.
+
+If a package is *trimmable* this means that a package's files will not be included in the application unless those files are directly referenced by another file or as a root.
+
+Additional *trimmable* files may be specified using the `TrimmableFiles` item.
+Additional *trimmable* packages may be specified using the `TrimmablePackages` item.
+
+### File relationships
+Managed assemblies are related to other managed assemblies by type references.  Managed assemblies are related to native libraries by DllImports, P-Invokes, to those libraries.
```

Comment:
While type references are correct it might be better to talk about assembly references.